src/jenkins_job_insight/jira.py

```
@@ -101,15 +102,109 @@ async def __aenter__(self) -> "JiraClient":
+                    if isinstance(p, dict) and "key" in p
+                ]
+                if query:
+                    _q = query.lower()
+                    result = [
+                        p
+                        for p in result
+                        if _q in p["key"].lower() or _q in p["name"].lower()
+                    ]
+                if result:
+                    return result
+            except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)